tests/e2e/operator/infra/gcp.go

```diff
@@ -0,0 +1,620 @@
+package infra
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/api/googleapi"
+
+	"github.com/cockroachdb/helm-charts/tests/e2e/coredns"
+	"github.com/cockroachdb/helm-charts/tests/e2e/operator"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/container/v1"
+	"google.golang.org/api/option"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// ─── GCP CONSTANTS ───────────────────────────────────────────────────────────────
+
+const (
+	projectID                = "cockroach-nishanth"
```

Comment:
I suggest making this configurable so you can run these in different projects (ex: in the new project we create for you or in your individual project to test changes).